### DISCUSSION TITLE
Route path need config the whole path

### DISCUSSION THREAD

**Discussion by XJShane** [Kong/kong#7715] [Help] [answered]
Route path need config the whole path when my request uri is: _**localhost:8000/api/login**_

route path need: _**/api/login**_

My **nginx.conf** is:
		location ^~ /api/ {
			default_type                     '';
					
			set $ctx_ref                     '';
			set $upstream_te                 '';
			set $upstream_host               '';
			set $upstream_upgrade            '';
			set $upstream_connection         '';
			set $upstream_scheme             '';
			set $upstream_uri                '';
			set $upstream_x_forwarded_for    '';
			set $upstream_x_forwarded_proto  '';
			set $upstream_x_forwarded_host   '';
			set $upstream_x_forwarded_port   '';
			set $upstream_x_forwarded_path   '';
			set $upstream_x_forwarded_prefix '';
			set $kong_proxy_mode             'http';
	
			proxy_http_version      1.1;
			proxy_buffering          on;
			proxy_request_buffering  on;
	
			proxy_set_header      TE                 $upstream_te;
			proxy_set_header      Host               $upstream_host;
			proxy_set_header      Upgrade            $upstream_upgrade;
			proxy_set_header      Connection         $upstream_connection;
			proxy_set_header      X-Forwarded-For    $upstream_x_forwarded_for;
			proxy_set_header      X-Forwarded-Proto  $upstream_x_forwarded_proto;
			proxy_set_header      X-Forwarded-Host   $upstream_x_forwarded_host;
			proxy_set_header      X-Forwarded-Port   $upstream_x_forwarded_port;
			proxy_set_header      X-Forwarded-Path   $upstream_x_forwarded_path;
			proxy_set_header      X-Forwarded-Prefix $upstream_x_forwarded_prefix;
			proxy_set_header      X-Real-IP          $remote_addr;
			proxy_pass_header     Server;
			proxy_pass_header     Date;
			proxy_ssl_name        $upstream_host;
			proxy_ssl_server_name on;
			proxy_pass            $upstream_scheme://kong_upstream$upstream_uri;
		}

I have already config the prefix `/api` in nginx.conf, why still i need to config the whole path(/api/login) in route，is that possible just config `/login` in route and also pass `/login` path to upstream?

Dese any body have this problem??? hope someone could help me out, thanks~

**Comment by fffonion** [comment, marked as answer]
@XJShane The location hack you did not change how Kong views the request path, but just point to a different location block, which does the same thing as the default `/` block. While the use case you described is not a common case in Kong, if you must
use it in that way, consider using an internal redirect (not rewrite).

For various historical reasons from #2366 to #6821, Kong won't use the URI after rewrite (it always uses the original request_uri), thus rewrite it in plugin or template doesn't work.
